### ISSUE TITLE
why not use enums for tree_criterion, parametric_method, selection_criterion?

### ISSUE THREAD

**Issue by slayoo** [vinecopulib/vinecopulib#158]

**Comment by tvatter** [comment]
That's a good idea.

The only "con" argument that I see is that it is more cumbersome to write e.g. tree_criterion::tau than just "tau".

**Comment by tnagler** [comment]
Agree. The "pro" argument is that the code expresses more clearly what "tau" is ;)

**Comment by tvatter** [comment]
Definitely, hence the quotes for "con" :)

All in favor say aye.

**Comment by tnagler** [comment]
Let's do it!

**Comment by tvatter** [comment]
@github Open a pull request to refactor the codebase and implement this idea

**Comment by tvatter** [comment]
Has been discussed and probably won't happen. Feel free to reopen sometimes.
